### PR TITLE
Making Tracking of how well are missions #10

### LINKED ISSUE #10: [Feature] Implement task tracking/warnings on home page
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
The app must keep track of the user's actions, tracking completed challenges and remembering the user on the home page to do challenges.

**Describe alternatives you've considered**
No.

**Additional context**
The design files are available at: https://xd.adobe.com/view/9be0bbe8-f524-435d-6702-efcf55fe51d7-edd1/grid

component (6)

component (7)

### PR DESCRIPTION
**Pull request by GustavoMottaTranche**
Makes a control of how much time the last mission was done and gives a feedback in color and text of the user is going good or not with that contact

### CODE REVIEW

File: src/app/home/home.page.ts [hunk 1]
```diff
@@ -50,23 +53,27 @@ export class HomePage implements OnInit {
             contactValue.mission = "Você fez " + missionsToday.toString() + " missões hoje.";
             contactValue.mission_label_color = "dark";
             contactValue.mission_color = "dark";
+            contactValue.color = '';
           }
           else {
-            contactValue.mission = daysAgoMissions.toString() + "Dias sem missão.";
-            contactValue.mission_label_color = "red";
-            contactValue.mission_color = "red";
+            contactValue.mission = daysAgoMissions.toString() + " Dias sem missão.";
```
Comment (vcasadei) [inline src/app/home/home.page.ts]: colocar " Dias..." em minúsculo

File: src/app/home/home.page.ts [hunk 2]
```diff
@@ -50,23 +53,27 @@ export class HomePage implements OnInit {
...
+            contactValue.mission_label_color = "danger";
+            contactValue.mission_color = "danger";
+            contactValue.color = '#eb445a';
           }
         }
         else {
           contactValue.mission = "Você fez " + missionsToday.toString() + " missões hoje.";
```
Comment (vcasadei) [inline src/app/home/home.page.ts:66]: Acredito que pode substituir o " missões hoje." por " missões hoje!"

File: src/app/pages/details/details.page.ts [hunk 3]
```diff
@@ -111,10 +111,10 @@ export class DetailsPage implements OnInit {
       if (this.person.challenges[i].id === challenge.id) {
         if (challenge.status == this.STATUS_CHECK) {
           this.person.challenges[i].status = this.STATUS_UNCHECK;
-
+          this.person.challenges[i].lastChange = new Date((new Date().getTime() - (3 * 3600 * 1000))).toISOString();
         } else {
           this.person.challenges[i].status = this.STATUS_CHECK;
-          this.person.challenges[i].lastChange = new Date().toISOString();
+          this.person.challenges[i].lastChange = new Date((new Date().getTime() - (3 * 3600 * 1000))).toISOString();
```
Comment (vcasadei) [inline src/app/pages/details/details.page.ts:117]: Does this fix the check/uncheck bug?

**Review by vcasadei — CHANGES_REQUESTED** [review]

File: src/app/pages/details/details.page.ts (hunk 3 above)
Comment (GustavoMottaTranche) [reply src/app/pages/details/details.page.ts:117]: No, what fixes the bug it's in home.page.ts in line 100, that is just to keep the tracking of updates in a challenge

**Pull request merged**